lib/core/diagnostics.js

```diff
@@ -114,6 +120,76 @@ if (undiciDebugLog.enabled || fetchDebuglog.enabled) {
   isClientSet = true
 }
 
+// Track fetch requests
+if (fetchDebuglog.enabled) {
+  const debuglog = fetchDebuglog.enabled
+
+  diagnosticsChannel.channel('undici:fetch:start').subscribe(evt => {
```

Comment:
Forgot to ask, sorry for circling back to this again, can we add a test for it?
It should be under `test/node-test/debug.js`

Reply:
For adding tests to the debug.js file, I added the extra tests for the additional logs, however, I noticed that I do not get a debug log for the error event, it is an empty string.....

Reply:
I also, see that the other debug logs do not check for the error log....

Reply:
what do you mean?

Reply:
So it looks like for my error debug log, in `fixtures/fetch.js` i am running into an error where res is undefined. Though I am not sure if I should be reaching this when it comes to the error debug logs. 

So in `fixtures/fetch.js` , `res => { res.body.cancel()}` returns an error 
```
undici/test/fixtures/fetch.js:14',
  '    res => { res.body.cancel()},',
  '                 ^',
  '',
  "TypeError: Cannot read properties of undefined (reading 'body')",
  '    at /test/fixtures/fetch.js:14:18',
  '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)',
  '',
  'Node.js v20.8.0',
  
```

Reply:
Yeah, the current tests are not aiming for error lines just yet; so feel free to just open one for the happy-path and we can continue on further PRs for the failure ones.

Though, not so sure how did you reach a happy path with an error, leading to res being `undefined` 🤔

Reply:
We already fixed it yesterday night ;).